prover/witness_generator/src/basic_circuits.rs

```diff
@@ -661,17 +703,50 @@ async fn generate_witness(
         }
     };
 
+    let mut eip_4844_blobs = eip_4844_blobs.blobs;
+
+    let mut eip_4844_circuits = vec![];
+    let mut eip_4844_witnesses = vec![];
+
+    let single_blob = eip_4844_blobs.len() == 1;
+    if single_blob {
+        eip_4844_blobs.push(eip_4844_blobs[0].clone());
+    }
+    for blob in eip_4844_blobs {
+        let (eip_4844_circuit, eip_4844_witness) = generate_eip4844_circuit_and_witness(
+            blob.to_vec(),
+            "/home/evl/zksync_era/trusted_setup.json",
+        );
+        eip_4844_circuits.push(eip_4844_circuit);
+        eip_4844_witnesses.push(eip_4844_witness.closed_form_input.observable_output);
+    }
```

Comment:
I prefer a declarative way: would something like `let (eip_4844_circuits, eip_4844_witnesses) = eip_4844_blobs.iter().map(generate_eip4844_circuit_and_witness).unzip()` work? It's a little hard to keep track on all the mutable lists here -when and why they are populated